app/src/main/java/com/example/android/sunshine/app/DetailFragment.java

```diff
@@ -151,62 +119,39 @@ private Intent createShareForecastIntent() {
 
     @Override
     public void onActivityCreated(Bundle savedInstanceState) {
-        getLoaderManager().initLoader(DETAIL_LOADER, null, this);
+        weatherDetailsLoader.startLoading();
         super.onActivityCreated(savedInstanceState);
     }
 
     void onLocationChanged(String newLocation) {
-        // replace the uri, since the location has changed
-        if (null != uri) {
-            long date = WeatherContract.WeatherEntry.getDateFromUri(uri);
-            uri = WeatherEntry.buildWeatherLocationWithDate(newLocation, date);
-            getLoaderManager().restartLoader(DETAIL_LOADER, null, this);
-        }
+        weatherDetailsLoader.restartLoadingFor(newLocation);
     }
 
-    @Override
-    public Loader<Cursor> onCreateLoader(int id, Bundle args) {
-        if (null != uri) {
-            // Now create and return a CursorLoader that will take care of
-            // creating a Cursor for the data being displayed.
-            return new CursorLoader(
-                    getActivity(),
-                    uri,
-                    DETAIL_COLUMNS,
-                    null,
-                    null,
-                    null
-            );
-        }
-        return null;
-    }
+    private OnWeatherDetailsLoadedListener onWeatherDetailsLoadedListener = new OnWeatherDetailsLoadedListener() {
+        @Override
+        public void onWeatherDetailsLoaded(WeatherConditions weatherConditions) {
+            if (weatherConditions != null) {
+                updateUiWith(weatherConditions);
+            }
 
-    @Override
-    public void onLoadFinished(Loader<Cursor> loader, Cursor data) {
-        if (data != null && data.moveToFirst()) {
-            updateUiWith(data);
+            setUpToolbar();
         }
+    };
 
-        setUpToolbar();
-    }
-
-    private void updateUiWith(Cursor data) {
-        // Read weather condition ID from cursor
-        int weatherId = data.getInt(COL_WEATHER_CONDITION_ID);
+    void updateUiWith(WeatherConditions conditions) {
+        int weatherId = conditions.weatherId();
 
         // Use weather art image
         Glide.with(this)
```

Comment:
I think it would be too much for this simple project. But I like the idea in general 👍